### PR TITLE
MD using updated libra library

### PR DESCRIPTION
**Pull request by Pulokesh**
Modified according to updated libra library and new python modules. MD simulation working but total energy is fluctuating with large magnitude - suspecting some bug in this version.

In the create_qe_input.py script, a and b are chaged to a-1 and b-1 as excite(a,b,gs_alp) function uses indices (N-1)

### CODE REVIEW

File: run/run.py [hunk 1]
```diff
@@ -37,16 +37,16 @@
 #params["nband"] = 12 #total number of orbitals
 #params["HOMO"] = 6   #orbital number of HOMO
 params["qe_debug_print"] = 1
-
+params["no_ex"] = 2
```
Comment (alexvakimov) [inline run/run.py]: please add a comment

File: src/create_qe_input.py [hunk 2]
```diff
@@ -56,17 +65,17 @@ def excitation_to_qe_occ(params, state):
             gs_bet.append([i,0.0])
 
     # Compute indices of the orbitals involved in the excitation
-    a = state.from_orbit[0] + homo
+    a = state.from_orbit[0] + homo -1 # As excite(a,b,gs_alp) uses indices (N-1)
```
Comment (alexvakimov) [inline src/create_qe_input.py]: there should be no "-1" - everything is already with indices. please double check

File: src/create_qe_input.py [hunk 3]
```diff
@@ -56,17 +65,17 @@ def excitation_to_qe_occ(params, state):
...
     a_s = state.from_spin[0]  # +1 for alp, -1 for bet
-    b = state.to_orbit[0] + homo
+    b = state.to_orbit[0] + homo -1   # As excite(a,b,gs_alp) uses indices (N-1)
```
Comment (alexvakimov) [inline src/create_qe_input.py]: same

File: src/main.py [hunk 4]
```diff
@@ -57,18 +58,21 @@ def main(params):
 
     params["cell_dm"], params["qe_inp_templ"] = read_qe_inp_templ(params["qe_inp0"])
 
-    exe_espresso(params["qe_inp0"], params["qe_out0"] )
-    tot_ene, label, R, grads = unpack_file(params["qe_out0"], params["qe_debug_print"])
+    exe_espresso(params["qe_inp0"], params["qe_out0"])
+    tot_ene, label, R, grads,data = unpack_file(params["qe_out0"], params, 0)
```
Comment (alexvakimov) [inline src/main.py]: please, get rid of the "data" dictionary. why is it needed?

File: src/unpack_file.py [hunk 5]
```diff
@@ -129,8 +128,8 @@ def unpack_file(filename,params, flag):
 
     nlines = len(A)
 
-    for a in A:
-        s = a.split()
+    for i in range(0,nlines):
+        s = A[i].split()  #is this A or a????
```
Comment (alexvakimov) [inline src/unpack_file.py]: please keep the original version

**Comment by alexvakimov** [comment]
Please, check that energy is conserved. And take care of other comments. Closing for now.

File: run/run.py (hunk 1 above)
Comment (Pulokesh) [inline run/run.py]: This is no longer needed. This is replaced by len(params["excitations"]).

File: src/create_qe_input.py (hunk 2 above)
Comment (Pulokesh) [inline src/create_qe_input.py]: As discussed today, we keep this unchanged and change HOMO = nocc_alp -1

File: src/create_qe_input.py (hunk 3 above)
Comment (Pulokesh) [inline src/create_qe_input.py]: kept unchanged.

File: src/main.py (hunk 4 above)
Comment (Pulokesh) [inline src/main.py]: data dictionary is removed now. norb and nel are returned in addition to those existing objects.

File: src/unpack_file.py (hunk 5 above)
Comment (Pulokesh) [inline src/unpack_file.py]: Original version is returned. In the following lines, icoord = A.index(a) and iforce = A.index(a) changes have been made.

**Comment by alexvakimov** [comment]
Ekadashi, please check the energy conservation and if it is fine, create another pull request

**Pull request closed without merging**